libqtile/ipc.py

```diff
@@ -43,11 +44,20 @@ class IPCError(Exception):
 
 class _IPC(object):
     def _unpack(self, data):
+
+        if data is None:
+            raise IPCError("received data is None")
+
+        try:
+            return json.loads(data.decode('utf-8')), True
+        except Exception:
```

Comment:
It might be better to figure out what exceptions malformed json will throw rather than catching bare Exceptions